### ISSUE TITLE
Create backing service to schedule ASG capacity

### ISSUE THREAD

**Issue by steinbachr** [MeanPug/folding-together#2]
Essentially, this service should understand how many blocks of time have been purchased and - based on this - adjust the ASG parameters. For instance, assume donor A purchased 20m of time and donor B 40m of time. The scheduler would then understand that an instance may be added to the ASG for 60m before a new scaling event. Assume then that a new donor purchases, 60m of time. Rather than line up blocks in serial, a new instance should be scheduled immediately and the ASG scaled up to 2 for 1 hour (rather than 1 for 2 hours).

The TlDr here is, schedule in 60m increments

**Comment by atkinsonm** [comment]
Linux provides per-second billing, so we have as much flexibility as we want for scheduling blocks of time.

That said, there's simplicity in performing a "balance lookup" every hour and maximizing capacity for that hour.

**Comment by steinbachr** [comment]
yeah, im all for simplicity in the 3 days we have to build 😄

**Comment by atkinsonm** [comment]
Comments on #3

**Comment by steinbachr** [comment]
how would you feel about running the scheduler and payment/user registration gateway in an EKS cluster? That would allow us to spin up a postgres image in-cluster rather than running a separate RDS instance. I was initially leaning towards using fargate clusters, but lack of persistence in the clusters is a non-starter (unless we use RDS instance). Might just be an easier workload to manage (having all our supporting services running in the same cluster)..

Thoughts?

**Comment by atkinsonm** [comment]
Cancelling based on plan to run one EC2 instance per donor. Eliminates the need for multi-donor ASG and complicated scheduling.
